components/nimbus/src/stateful/evaluator.rs

```diff
@@ -53,4 +57,45 @@ impl TargetingAttributes {
         self.days_since_update = update_date.map(|then| (now - then).num_days() as i32);
         self.current_date = now;
     }
+
+    pub(crate) fn update_enrollments(&mut self, enrollments: Vec<ExperimentEnrollment>) -> u32 {
+        let mut modified_count = 0;
+        for ee in enrollments {
+            if self.update_enrollment(ee) {
+                modified_count += 1;
+            }
+        }
+        modified_count
+    }
+
+    pub(crate) fn update_enrollment(&mut self, enrollment: ExperimentEnrollment) -> bool {
```

Comment:
This does not need to take an owned ExperimentEnrollment from what I can see. This should take a `&ExperimentEnrollment`. Then we don't need to clone() the enrollments on the way in.